src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemDisplay.cs

```diff
@@ -105,37 +104,7 @@ private ReadOnlyCollection<DescriptionItem> CreateDescriptionItems()
         public string Name => _searchResult.NavigableItem.DisplayTaggedParts.JoinText();
 
         public void NavigateTo()
-        {
-            var token = _asyncListener.BeginAsyncOperation(nameof(NavigateTo));
-            NavigateToAsync().ReportNonFatalErrorAsync().CompletesAsyncOperation(token);
-        }
-
-        private async Task NavigateToAsync()
-        {
-            var document = _searchResult.NavigableItem.Document;
-            if (document == null)
-                return;
-
-            var workspace = document.Project.Solution.Workspace;
-            var navigationService = workspace.Services.GetService<IDocumentNavigationService>();
-
-            // Document tabs opened by NavigateTo are carefully created as preview or regular tabs
-            // by them; trying to specifically open them in a particular kind of tab here has no
-            // effect.
-            //
-            // In the case of a stale item, don't require that the span be in bounds of the document
-            // as it exists right now.
-            using var context = _threadOperationExecutor.BeginExecute(
-                EditorFeaturesResources.Navigating_to_definition, EditorFeaturesResources.Navigating_to_definition, allowCancellation: true, showProgress: false);
-            await navigationService.TryNavigateToSpanAsync(
-                _threadingContext,
-                workspace,
-                document.Id,
-                _searchResult.NavigableItem.SourceSpan,
-                NavigationOptions.Default,
-                allowInvalidSpan: _searchResult.NavigableItem.IsStale,
-                context.UserCancellationToken).ConfigureAwait(false);
-        }
+            => NavigateToHelpers.NavigateTo(_searchResult, _threadingContext, _threadOperationExecutor, _asyncListener);
```

Comment:
moved to helper so we can use the same navigation code in go-to and the new aiosp.